src/auth.ts

```diff
@@ -548,29 +553,73 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
   }
 }
 
+/**
+ * hidden
+ */
 export abstract class NativeAuthType {
   protected config: IConfig;
   protected client: IClient;
+  protected userService: ISingleUserService;
+  private tokenContext: ICombinedTokenContext;
+  private emitter: IEventEmitter;
+  private authToken: string;
 
-  constructor(deps: AuthTypeDependencies) {
+  constructor(deps: NativeAuthDependencies) {
     this.config = deps.config;
     this.client = deps.client;
+    this.userService = deps.userService;
+    this.tokenContext = deps.tokenContext;
+    this.emitter = deps.emitter;
   }
+
+  /**
+   * Get the raw auth token of the active user from local storage.
+   */
+  public getToken(): string {
+    return this.tokenContext.get();
+  }
+
+  /**
+   * @hidden
+   */
+  public storeToken(token: string) {
+    let originalToken = this.authToken;
+    this.authToken = token;
+    this.tokenContext.store(this.authToken, {'permanent': true});
+    this.emitter.emit('auth:token-changed', {'old': originalToken, 'new': this.authToken});
+  }
+
 }
 
-export class GoogleNativeAuth extends NativeAuthType {
-  public authenticate(data?: any, options?: any): Promise<any> {
+/**
+ * GoogleNativeAuth handles logging into googleplus through the cordova-plugin-googleplus plugin.'
+ * @featured
+ */
+export class GoogleAuth extends NativeAuthType implements IGoogleAuth {
+  public login(scope: GoogleScopes[]): Promise<any> {
     let deferred = new DeferredPromise<any, Error>();
-    const clientID = this.config.settings.nativeAuth['webClientId'];
+    const clientID = this.config.settings.auth.google.webClientId;
 
     if (!clientID) {
       deferred.reject(new Error('Missing google web client id. Please visit http://docs.ionic.io/services/users/google-auth.html#native'));
     }
 
-    GooglePlus.login({'webClientId': clientID, 'offline': true}).then((success) => {
-      const request_object = {
+    // Reqire basic profile data.
+    if (scope.indexOf('profile') === -1) {
```

Comment:
You might be able to use ES6 sets to clean up this logic: http://www.2ality.com/2015/01/es6-maps-sets.html#set

Although I'm not sure if we want to include shims for sets, yet.